env/file_system_tracer.h

```diff
@@ -0,0 +1,319 @@
+#pragma once
+
+#include "rocksdb/file_system.h"
+#include "trace_replay/io_tracer.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+// FileSystemTracingWrapper is a wrapper class above FileSystem that forwards
+// the call to the underlying storage system. It then invokes IOTracer to record
+// file operations and other contextual information in a binary format for
+// tracing. It overrides methods we are interested in tracing and extends
+// FileSystemWrapper, which forwards all methods that are not explicitly
+// overridden.
```

Comment:
In order to use the IO tracer, application need to pass FileSystemTracingWrapper to the options.env?

Reply:
No they dont't. FileSystemPtr  overload operator -> and will check if tracing is enabled or not and return pointer to Filesystem or FileSystemTracingWrapper based on that. I will share the design doc with you.  Thank you for taking a look.